app/screens/reaction_list/reaction_list.js

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import {intlShape} from 'react-intl';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {
+    generateUserProfilesById,
+    getMissingUserIds,
+    getReactionsByName,
+    getSortedReactionsForHeader,
+    getUniqueUserIds,
+    sortReactions,
+} from 'app/utils/reaction';
+
+import ReactionHeader from './reaction_header';
+import ReactionRow from './reaction_row';
+
+import {ALL_EMOJIS} from 'app/constants/emoji';
+
+export default class ReactionList extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            getMissingProfilesByIds: PropTypes.func.isRequired,
+        }).isRequired,
+        navigator: PropTypes.object,
+        reactions: PropTypes.array.isRequired,
+        theme: PropTypes.object.isRequired,
+        teammateNameDisplay: PropTypes.string,
+        userProfiles: PropTypes.array,
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        const {reactions, userProfiles} = props;
+
+        const reactionsByName = getReactionsByName(reactions);
+
+        this.state = {
+            allUserIds: getUniqueUserIds(reactions),
+            reactions,
+            reactionsByName,
+            selected: ALL_EMOJIS,
+            sortedReactions: sortReactions(reactionsByName),
+            sortedReactionsForHeader: getSortedReactionsForHeader(reactionsByName),
+            userProfiles,
+            userProfilesById: generateUserProfilesById(userProfiles),
+        };
+
+        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        let hasNewState = false;
+        const newState = {};
+        if (!nextProps.reactions.length !== prevState.reactions.length) {
```

Comment:
Instead of comparing lengths, can't this just be `nextProps.reactions !== prevState.reactions`? Also, you appear to have an extra `!` here anyway, so if a post has any reactions, it's comparing `false !== a number`

Reply:
Oh thanks, that's from my initial iteration and missed to clean it up.